### DISCUSSION TITLE
Is there a way to reset the stubPath in a subdirectory?

### DISCUSSION THREAD

**Discussion by Gobot1234** [microsoft/pyright#4850] [Q&A] [answered]
As background I'm working on a project for a Arduino Nano which runs micropython and has a slightly different stdlib to cpython. I'm using a custom typeshed for the project (https://github.com/hlovatt/PyBoardTypeshed) and moving these files into a typings file so that development inside of the "core" folder, which is meant to be uploaded to the Ardunio, is much nicer and fully typed. 
```
├── core
│   ├── __init__.py
│   └── main.py
├── scripts
│   ├── build.py
│   ├── install.py
│   ├── pyboard.py
│   └── pyrightconfig.json
└── typings
    ├── _thread.pyi
    ├── array.pyi
 ...

7 directories, 74 files
```
My issue is that the "scripts" directory is meant to be ran on my computer not the Arduino and so has the standard typeshed. Is there any way to have the "scripts" directory to use the default typeshed whilst keeping my "typings" for use in core?

**Comment by erictraut** [comment, marked as answer]
Are you using the pyright CLI or a language server? If it's the CLI, you can create two different config files and run pyright twice, once for each subproject. If you're using a language server and your client supports multi-root workspaces, you can create a different workspace root for each project. Both of these allow you to specify different stubPath or typeshedPath settings for different subprojects.

**Reply by Gobot1234** [reply]
Multi-root workspaces worked like a charm, thanks.
